src/table/utils/selections-utils.js

```diff
@@ -81,7 +81,12 @@ export const getSelectedRows = ({ selectedRows, cell, evt }) => {
     return { [qElemNumber]: rowIdx };
   }
 
-  if (selectedRows[qElemNumber] !== undefined) {
+  if (evt.shiftKey && ((evt.key === 'ArrowUp' && cell.rowIdx !== 0) || evt.key === 'ArrowDown')) {
```

Comment:
no need to check the row index twice. this should be enough
```suggestion
  if (evt.shiftKey && evt.key.includes('Arrow') {
```